grafana/grafana
# Users: clean up OrgId when all user orgs are removed

**What this PR does / why we need it**:

This is to fix #20224. With the fix, when all user orgs are removed we now set `OrgId` to `0` and omit it from the `/api/users/:id` response.

**Which issue(s) this PR fixes**:



Fixes #20224

**Special notes for your reviewer**:

How to test?

Add a new user and check the user id:

<img width="848" alt="Screen Shot 2022-03-01 at 12 36 11" src="https://user-images.githubusercontent.com/745532/156162363-4697d01d-c201-470f-9f86-62b97c1c5d6f.png">

Go to `http://localhost:3000/api/users` + `user id` (e.g. http://localhost:3000/api/users/4). The response should contain automatically assigned `orgId`:

```json
{
  "id": 4,
  "email": "user@user.user",
  ...
  "orgId": 1,
}
```

Remove the user from the organization and check the response from `/api/users/...`. It should not contain `orgId` any more.
